configs/components/ruby-shadow.rb

```diff
@@ -6,42 +6,14 @@
   pkg.environment "PATH", "$(PATH):/usr/ccs/bin:/usr/sfw/bin"
   pkg.environment "CONFIGURE_ARGS", '--vendor'
 
-  if platform.is_solaris?
-    if platform.architecture == 'sparc'
-      pkg.environment "RUBY", settings[:host_ruby]
-    end
-    ruby = "#{settings[:host_ruby]} -r#{settings[:datadir]}/doc/rbconfig-#{settings[:ruby_version]}-orig.rb"
-  elsif platform.is_cross_compiled?
+  if platform.is_cross_compiled?
```

Comment:
In theory we shouldn't preload `rbconfig-xxx-orig.rb` when doing native compiles on Solaris. But the `agent-runtime-7.x` fails on `solaris-1*-i386` without it:

```
12:15:38 export PATH="/usr/sbin:/usr/bin:/usr/ccs/bin:/usr/sfw/bin" && \
12:15:38 export CONFIGURE_ARGS="--vendor"
12:15:38 touch ruby-shadow-unpack
12:15:38 touch ruby-shadow-patch
12:15:38 [ -d ruby-shadow ] || mkdir -p ruby-shadow
12:15:38 touch ruby-shadow-configure
12:15:38 export PATH="/usr/sbin:/usr/bin:/usr/ccs/bin:/usr/sfw/bin" && \
12:15:38 export CONFIGURE_ARGS="--vendor" && \
12:15:38 cd ruby-shadow && \
12:15:38 /opt/puppetlabs/puppet/bin/ruby extconf.rb && \
12:15:38 gmake -e -j3
12:15:38 checking for getspent() in -lshadow... *** extconf.rb failed ***
12:15:38 Could not create Makefile due to some reason, probably lack of necessary
12:15:38 libraries and/or headers.  Check the mkmf.log file for more details.  You may
12:15:38 need configuration options.
12:15:38 
12:15:38 Provided configuration options:
12:15:38 	--with-opt-dir
12:15:38 	--with-opt-include
12:15:38 	--without-opt-include=${opt-dir}/include
12:15:38 	--with-opt-lib
12:15:38 	--without-opt-lib=${opt-dir}/lib
12:15:38 	--with-make-prog
12:15:38 	--without-make-prog
12:15:38 	--srcdir=.
12:15:38 	--curdir
12:15:38 	--ruby=/opt/puppetlabs/puppet/bin/$(RUBY_BASE_NAME)
12:15:38 	--with-shadow-dir
12:15:38 	--without-shadow-dir
12:15:38 	--with-shadow-include
12:15:38 	--without-shadow-include=${shadow-dir}/include
12:15:38 	--with-shadow-lib
12:15:38 	--without-shadow-lib=${shadow-dir}/lib
12:15:38 	--with-shadowlib
12:15:38 	--without-shadowlib
12:15:38 /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:471:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
12:15:38 You have to install development tools first.
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:564:in `try_link0'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:582:in `try_link'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:794:in `try_func'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:1029:in `block in have_library'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:971:in `block in checking_for'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:361:in `block (2 levels) in postpone'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:331:in `open'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:361:in `block in postpone'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:331:in `open'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:357:in `postpone'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:970:in `checking_for'
12:15:38 	from /opt/puppetlabs/puppet/lib/ruby/2.7.0/mkmf.rb:1024:in `have_library'
12:15:38 	from extconf.rb:31:in `<main>'
12:15:38 gmake: *** [ruby-shadow-build] Error 1
```

I think the problem is `/opt/pl-build-tools/bin` is missing from PATH

Reply:
Yeah adding to `PATH` works:
```
/opt/puppetlabs/puppet/bin# uname -a
SunOS asocial-uptake 5.11 11.1 i86pc i386 i86pc
/opt/puppetlabs/puppet/bin# ./irb
irb(main):001:0> require 'shadow'
=> true
irb(main):002:0> Shadow::Passwd.getspent
=> #<struct Struct::PasswdEntry sp_namp="root", sp_pwdp=<redacted>>
```